### ISSUE TITLE
Dataframe expression explanations sould be more prominent

### ISSUE THREAD

**Issue by savente93** [nushell/nushell.github.io#1084]
Currently I find the documentation around dataframes and dataframe expressions to be very unclear/difficult to navigate. Some examples:

1. In the documentation `commands` tab, there is a section labeled `expression`. This is about dataframe expressions specifically, and lazy data frames at that from as far as I can tell. I expected this tab to be about nu expressions specifically. This also meant that I thought a lot of documentation around dataframes such as `group-by` was missing, while it's actually in the `expression` section.
2. The difference between lazy and eager dataframes is exlained but the implications of this are mentioned only off hand. For example, the fact that you can only  use `dfr col` in a lazy expression, but `dfr as datetime` can't be at all was extremely confusing to me. 
3. Having three seperate pages for `dfr`commands (dataframe, dataframe or lazy dataframe and expression) makes it much harder to find things and is also bad for discoverability. I for example, just thought a bunch of standard things in dataframes were not implemented because of this.

These are the issues that come to mind, I hope this will be read as constructive, as that's how I mean it, not just to criticise. These were just a few issues that caused me a lot of confusion when trying to understand them. Thanks for all the hard work!

**Comment by fdncred** [comment]
Agreed. If there aren't examples in the commands, I usually can't figure it out myself and have to ask around in our #dataframes discord channel.
